include/preprocess/svd_rounding_step.hpp

```diff
@@ -0,0 +1,116 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+
+//Contributed and/or modified by Alexandros Manochis, as part of Google Summer of Code 2020 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+
+#ifndef SVD_ROUNDING_SINGLE_STEP_HPP
+#define SVD_ROUNDING_SINGLE_STEP_HPP
+
+
+template
+<
+    typename WalkTypePolicy,
+    typename MT,
+    typename VT,
+    typename Polytope,
+    typename Point,
+    typename NT,
+    typename svd_params,
+    typename RandomNumberGenerator
+>
+
+void svd_rounding_single_step(Polytope &P,
+                              std::pair<Point,NT> &InnerBall,
+                              const unsigned int &walk_length,
+                              svd_params &parameters,
+                              RandomNumberGenerator &rng)
+{
+    //NT tol = 0.00000001;
+    //NT R = std::pow(10,10), r = InnerBall.second;
+
+    int n = P.dimension();//, m = P.num_of_hyperplanes();
+
+    //Polytope P_old(P);
+
+    MT round_mat, r_inv;
+    VT shift(n), s(n);
+
+    Point p(n);
+
+    //bool last_round_under_p;
+
+    //unsigned int round_it;
+    NT num_its, prev_max_s = std::numeric_limits<NT>::max(),
```

Comment:
please initialize one var at a time, it is more clear, here you use 3 initialization values.